executor/cook/executor.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+
+import sys
+import json
+
+from threading import Event, Thread
+
+from mesos.executor import MesosExecutorDriver
+from mesos.interface import Executor, mesos_pb2
+
+from server import run_server
+from store import Store, WATCH_ACTION_PUT
+from command import Command, run_commands
+
+def new_task_info(task, state, data = None):
+    t = mesos_pb2.TaskStatus()
+    t.task_id.value = task.task_id.value
+    t.state = state
+    if data:
+        t.data = json.dumps(data, separators=(',', ':'))
+
+    return t
+
+class CookExecutor(Executor):
+    def __init__(self):
+        self.store = Store({
+            'task': {
+                'reason': str,
+                'progress': float,
+                'env': {str: str}
+            }})
+        self.stop_event = Event()
+        self.server_thread = None
+        self.commands_thread = None
+
+    def registered(self, driver, executorInfo, frameworkInfo, slaveInfo):
+        print "CookExecutor registered"
```

Comment:
@wyegelwel I realized after starting to go from python 2 to 3 that the libmesos python bindings only work with python 2 (https://issues.apache.org/jira/browse/MESOS-355). If supporting python 3 is important, we'll have to start using the mesos agent HTTP API. What do you think?